### PR TITLE
PIN-2022: add GET -> /producer/eservices/{eserviceId}/descriptor/{descriptorId}

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -1481,6 +1481,122 @@ paths:
...
+              description: Max allowed requests within time interval
+            'X-Rate-Limit-Remaining':
+              schema:
+                type: integer
+              description: Remaining requests within time interval
+            'X-Rate-Limit-Interval':
+              schema:
+                type: integer
+              description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+        '400':
```
Comment (beetlecrunch) [inline src/main/resources/interface-specification.yml]: This is used instead of a 403 to reduce the permission visibility.
Tell me your opinion about it.

Comment (galales) [reply src/main/resources/interface-specification.yml]: is there a the need in this case to reduce this visibility?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 2]
```diff
@@ -248,6 +246,65 @@ final case class EServicesApiServiceImpl(
...
+    )
+
+    onComplete(result) {
+      handleError(s"Error retrieving producer descriptor $descriptorId of eservice $eserviceId") orElse {
+        case Failure(x: EServiceDescriptorNotFound) =>
+          logger.warn(x.getMessage)
+          getProducerEServiceDescriptor404(problemOf(StatusCodes.NotFound, x))
+        case Failure(x: InvalidEServiceRequester)   =>
+          logger.error(x.getMessage)
+          getProducerEServiceDescriptor400(problemOf(StatusCodes.NotFound, x))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]: status codes do not match

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 3]
```diff
@@ -248,6 +246,65 @@ final case class EServicesApiServiceImpl(
...
+  override def getProducerEServiceDescriptor(eserviceId: String, descriptorId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerProducerDescriptor: ToEntityMarshaller[ProducerEServiceDescriptor]
+  ): Route = {
+    val result: Future[ProducerEServiceDescriptor] = for {
+      requesterId                    <- getOrganizationIdFutureUUID(contexts)
+      (eserviceUUID, descriptorUUID) <- eserviceId.toFutureUUID.zip(descriptorId.toFutureUUID)
+      eService                       <- catalogProcessService.getEServiceById(eserviceUUID)
+      _                              <- isTheProducer(eService, requesterId)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala:259]: Captain Obvious 🎖️  here: this is an antipattern, even though I do not currently see any alternative

File: src/main/resources/interface-specification.yml (hunk 1 above)
Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml]: No. Moving to 403.

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala (hunk 3 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala:259]: let's say we bend this authorization not to the data returned but the access to the endpoint, and leave it as is

**Review by galales — APPROVED** [review]

**Pull request merged**